internal/service/base/resource_language_translation.go

```diff
@@ -0,0 +1,507 @@
+		state.TranslatedText = types.StringValue(translation.TranslatedText)
+	}
+
+	return nil
+}
+
+func (r *languageTranslationResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data languageTranslationResourceModel
+
+	if r.Client == nil || r.Client.ManagementAPIClient == nil {
+		resp.Diagnostics.AddError(
+			"Client not initialized",
+			"Expected the PingOne client, got nil.  Please report this issue to the provider maintainers.")
+		return
+	}
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update API call logic, since this is a singleton resource
+	clientData, diags := data.buildClientStruct()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var responseData []management.LocaleTranslation
+	resp.Diagnostics.Append(framework.ParseResponse(
+		ctx,
+
+		func() (any, *http.Response, error) {
+			fO, fR, fErr := r.Client.ManagementAPIClient.TranslationsApi.UpdateTranslations(ctx, data.EnvironmentId.ValueString(), data.Locale.ValueString()).LocaleTranslation(*clientData).Execute()
+
+			return framework.CheckEnvironmentExistsOnPermissionsError(ctx, r.Client.ManagementAPIClient, data.EnvironmentId.ValueString(), fO, fR, fErr)
+		},
+		"UpdateTranslations-Create",
+		framework.DefaultCustomError,
+		sdk.DefaultCreateReadRetryable,
+		&responseData,
+	)...)
+
+	// Subsequent read needed as the API does not return the full object on any create or update operation.
+	resp.Diagnostics.Append(framework.ParseResponse(
+		ctx,
+
+		func() (any, *http.Response, error) {
+			pagedIterator := r.Client.ManagementAPIClient.TranslationsApi.ReadTranslations(ctx, data.EnvironmentId.ValueString(), data.Locale.ValueString()).Execute()
+			var initialHttpResponse *http.Response
+
+			for pageCursor, err := range pagedIterator {
+				if err != nil {
+					return framework.CheckEnvironmentExistsOnPermissionsError(ctx, r.Client.ManagementAPIClient, data.EnvironmentId.ValueString(), nil, pageCursor.HTTPResponse, err)
+				}
+
+				if initialHttpResponse == nil {
+					initialHttpResponse = pageCursor.HTTPResponse
+				}
+
+				if translations, ok := pageCursor.EntityArray.Embedded.GetTranslationsOk(); ok {
+					for _, translation := range translations {
+						if v, ok := translation.GetKeyOk(); ok && *v == data.Key.ValueString() {
+							localeTranslationSlice := []management.LocaleTranslation{}
+							localeTranslationSlice = append(localeTranslationSlice, translation)
+							return localeTranslationSlice, pageCursor.HTTPResponse, nil
+						}
+					}
+				}
+			}
+
+			return nil, initialHttpResponse, nil
+		},
+		"UpdateTranslations-Create-SubsequentRead",
+		framework.CustomErrorResourceNotFoundWarning,
+		sdk.DefaultCreateReadRetryable,
+		&responseData,
+	)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Read response into the model
```

Comment:
I'd add some validation here that responseData has exactly 1 element as expected - same for update